apps/faucet/src/endpoints/impersonate/impersonate.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Address } from "@multiversx/sdk-core/out";
+import { Injectable } from "@nestjs/common";
+
+@Injectable()
+export class ImpersonateService {
+  async isImpersonateAllowed(address: string, impersonator: string): Promise<boolean> {
+    if (!this.isAddress(address)) {
+      return false;
+    }
+
+    if (!this.isAddress(impersonator)) {
+      return false;
+    }
+
+    return new Address(impersonator).isContractAddress();
+  }
+
+  private isAddress(address: string): boolean {
```

Comment:
maybe you can include the sdk-nestjs and use the AddressUtils.isAddressValid(address)

Reply:
removed